receiver/flinkmetricsreceiver/metadata.yaml

```diff
@@ -0,0 +1,319 @@
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+      input_type: string
+    attributes: []
+  flink.jvm.memory.non_heap.max:
+    enabled: true
+    description: The maximum amount of non-heap memory that can be used for memory management.
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+      input_type: string
+    attributes: []
+  flink.jvm.memory.metaspace.used:
+    enabled: true
+    description: The amount of memory currently used in the Metaspace memory pool.
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+      input_type: string
+    attributes: []
+  flink.jvm.memory.metaspace.committed:
+    enabled: true
+    description: The amount of memory guaranteed to be available to the JVM in the Metaspace memory pool.
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+      input_type: string
+    attributes: []
+  flink.jvm.memory.metaspace.max:
+    enabled: true
+    description: The maximum amount of memory that can be used in the Metaspace memory pool.
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+      input_type: string
+    attributes: []
+  flink.jvm.memory.direct.used:
+    enabled: true
+    description: The amount of memory used by the JVM for the direct buffer pool.
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+      input_type: string
+    attributes: []
+  flink.jvm.memory.direct.total_capacity:
+    enabled: true
+    description: The total capacity of all buffers in the direct buffer pool.
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+      input_type: string
+    attributes: []
+  flink.jvm.memory.mapped.used:
+    enabled: true
+    description: The amount of memory used by the JVM for the mapped buffer pool.
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+      input_type: string
+    attributes: []
+  flink.jvm.memory.mapped.total_capacity:
+    enabled: true
+    description: The number of buffers in the mapped buffer pool.
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+      input_type: string
+    attributes: []
+  flink.memory.managed.used:
+    enabled: true
+    description: The amount of managed memory currently used.
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+      input_type: string
+    attributes: []
+  flink.memory.managed.total:
+    enabled: true
+    description: The total amount of managed memory.
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+      input_type: string
+    attributes: []
```

Comment:
Is this following an established pattern of metrics? If so, can you link me to that design?

If not, can any of these be combined?

Reply:
To be honest, I think the design of these metrics is flawed and will end up being changed before any component using them is considered stable.

That said, we can accept that these are _useful_ without going through the entire semantic conventions discussion at this point. At least though, please change `non_heap` to `nonheap`, to match the linked metrics.

Reply:
updated to nonheap